### ISSUE TITLE
Error in reconstruct-counts

### ISSUE THREAD

**Issue by luhugerth** [jwdebelius/q2-sidle#5]
Running the reconstruction in the tutorial (won't copy the full command here), I get the error
```
Plugin error from sidle:

  "['max-mismatch'] not in index"

Debug info has been saved to /tmp/qiime2-q2cli-err-2kxnpvvr.log
```

The log file is as follows:
```
Traceback (most recent call last):
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.8/lib/python3.6/site-packages/q2cli/commands.py", line 329, in __call__
    results = action(**arguments)
  File "<decorator-gen-173>", line 2, in reconstruct_counts
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.8/lib/python3.6/site-packages/qiime2/sdk/action.py", line 231, in bound_callable
    for a in user_input[name]]
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.8/lib/python3.6/site-packages/qiime2/sdk/action.py", line 231, in <listcomp>
    for a in user_input[name]]
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.8/lib/python3.6/site-packages/qiime2/sdk/result.py", line 289, in _view
    result = transformation(self._archiver.data_dir)
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.8/lib/python3.6/site-packages/qiime2/core/transform.py", line 70, in transformation
    new_view = transformer(view)
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.8/lib/python3.6/site-packages/qiime2/core/transform.py", line 214, in wrapped
    return transformer(view.file.view(self._wrapped_view_type))
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.8/lib/python3.6/site-packages/q2_sidle/_transformer.py", line 57, in _5
    df = df[['kmer', 'asv', 'length', 'mismatch', 'max-mismatch', 'region']]
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.8/lib/python3.6/site-packages/pandas/core/frame.py", line 3001, in __getitem__
    indexer = self.loc._convert_to_indexer(key, axis=1, raise_missing=True)
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.8/lib/python3.6/site-packages/pandas/core/indexing.py", line 1285, in _convert_to_indexer
    return self._get_listlike_indexer(obj, axis, **kwargs)[1]
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.8/lib/python3.6/site-packages/pandas/core/indexing.py", line 1092, in _get_listlike_indexer
    keyarr, indexer, o._get_axis_number(axis), raise_missing=raise_missing
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.8/lib/python3.6/site-packages/pandas/core/indexing.py", line 1185, in _validate_read_indexer
    raise KeyError("{} not in index".format(not_found))
KeyError: "['max-mismatch'] not in index"
```

**Comment by luhugerth** [comment]
Tried again, now in Python 2.7.18, since this version could solve my previous issue. However:

 ```
Plugin error from sidle:

  "Passing list-likes to .loc or [] with any missing labels is no longer supported. The following labels were missing: Index(['4354577'], dtype='object', name='db-seq'). See https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#deprecate-loc-reindex-listlike"

Debug info has been saved to /tmp/qiime2-q2cli-err-xi5ypxyh.log
```

And the log file reads

```Regional Alignments Loaded
Traceback (most recent call last):
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.11/lib/python3.6/site-packages/q2cli/commands.py", line 329, in __call__
    results = action(**arguments)
  File "<decorator-gen-173>", line 2, in reconstruct_counts
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.11/lib/python3.6/site-packages/qiime2/sdk/action.py", line 245, in bound_callable
    output_types, provenance)
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.11/lib/python3.6/site-packages/qiime2/sdk/action.py", line 390, in _callable_executor_
    output_views = self._callable(**view_args)
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.11/lib/python3.6/site-packages/q2_sidle/_reconstruct.py", line 120, in reconstruct_counts
    kmer_map = kmer_map.loc[kmers]
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.11/lib/python3.6/site-packages/pandas/core/indexing.py", line 879, in __getitem__
    return self._getitem_axis(maybe_callable, axis=axis)
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.11/lib/python3.6/site-packages/pandas/core/indexing.py", line 1099, in _getitem_axis
    return self._getitem_iterable(key, axis=axis)
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.11/lib/python3.6/site-packages/pandas/core/indexing.py", line 1037, in _getitem_iterable
    keyarr, indexer = self._get_listlike_indexer(key, axis, raise_missing=False)
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.11/lib/python3.6/site-packages/pandas/core/indexing.py", line 1254, in _get_listlike_indexer
    self._validate_read_indexer(keyarr, indexer, axis, raise_missing=raise_missing)
  File "/home/luisa.hugerth/.conda/envs/qiime2-2020.11/lib/python3.6/site-packages/pandas/core/indexing.py", line 1316, in _validate_read_indexer
    "Passing list-likes to .loc or [] with any missing labels "
KeyError: "Passing list-likes to .loc or [] with any missing labels is no longer supported. The following labels were missing: Index(['4354577'], dtype='object', name='db-seq'). See https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#deprecate-loc-reindex-listlike"
```

**Comment by jwdebelius** [comment]
Also fixed in #9. The `max_mismatch` is inherited from alignment which works better with different sequencing lengths and mismatch probabilities
